File: src/main/java/net/okocraft/respawner/Respawner.java
package net.okocraft.respawner;

import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.PluginCommand;
import org.bukkit.command.TabCompleter;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.util.StringUtil;

public class Respawner extends JavaPlugin implements CommandExecutor, TabCompleter {

	private FileConfiguration config;
	private FileConfiguration defaultConfig;

	@Override
	public void onEnable() {
		PluginCommand command = Objects.requireNonNull(getCommand("respawner"));
		command.setExecutor(this);
		command.setTabCompleter(this);
		saveDefaultConfig();

		config = getConfig();
		defaultConfig = getDefaultConfig();
	}

	@Override
	public void onDisable() {
	}

	@Override
	public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
		if (args.length == 0) {
			sender.sendMessage(getMessage("specify-player"));
			return false;
		}

		Player target = Bukkit.getPlayer(args[0]);
		if (target == null) {
			sender.sendMessage(getMessage("player-not-found"));
			return false;
		}

		if (!target.isDead()) {
			sender.sendMessage(getMessage("player-is-alive"));
			return false;
		}

		target.spigot().respawn();
		sender.sendMessage(getMessage("respawned-player"));
		target.sendMessage(getMessage("respawned-by"));
		return true;
	}

	@Override
	public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
		if (args.length == 1) {
			return StringUtil.copyPartialMatches(args[0], Bukkit.getOnlinePlayers().stream().map(Player::getName).collect(Collectors.toSet()), new ArrayList<>());
		}
		return List.of();
	}

	private FileConfiguration getDefaultConfig() {
		InputStream is = getResource("config.yml");
		return YamlConfiguration.loadConfiguration(new InputStreamReader(is));
	}

	private String getMessage(String key) {
		String fullKey = "messages." + key;
		return config.getString(fullKey, defaultConfig.getString(fullKey));
	}
}

File: src/main/kotlin/net/okocraft/respawner/Respawner.kt
package net.okocraft.respawner

import org.bukkit.plugin.java.JavaPlugin

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.util.StringUtil

import hazae41.minecraft.kutils.bukkit.command
import hazae41.minecraft.kutils.bukkit.info
import hazae41.minecraft.kutils.bukkit.msg

class Respawner: JavaPlugin(), TabCompleter {
    override fun onEnable() {
        command("respawn") { sender, args ->
            if (args.isEmpty()) {
                sender.msg("&7* プレイヤーを指定してください。")

                return@command
            }

            val request = args[0]
            val target = Bukkit.getPlayer(request) ?: run {
                sender.msg("&7* &b$request&7 は存在しません。")

                return@command
            }

            if (target.isDead) {
                target.spigot().respawn()
                sender.msg("&7* &b${target.name}&7 を復活させました。")
                target.msg("&7* あなたは &b${sender.name}&7 によって復活させられました。")

                return@command
            }

            sender.msg("&7* &b${target.name}&7 は生きています。")
        }

        info("Respawner is enabled.")
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        alias: String,
        args: Array<out String>
    ): MutableList<String> {
        if (args.size == 1) {
            val dead = Bukkit.getOnlinePlayers()
                .filter { it.isDead }
                .map { it.name }
                .toMutableList()

            return StringUtil.copyPartialMatches(
                args[0], dead, mutableListOf()
            )
        }

        return mutableListOf()
    }
}